src/Akeneo/Pim/Enrichment/Component/Product/Connector/Processor/Normalization/ProductProcessor.php

```diff
@@ -149,9 +149,15 @@ protected function filterValues(array $values, array $attributesToFilter)
     {
         $valuesToExport = [];
         $attributesToFilter = array_flip($attributesToFilter);
-        foreach ($values as $code => $value) {
-            if (isset($attributesToFilter[$code])) {
-                $valuesToExport[$code] = $value;
+        foreach ($values as $attributeCode => $value) {
+            $attribute = $this->attributeRepository->findOneByIdentifier($attributeCode);
+            if ($attribute->isLocaleSpecific()) {
+                $availableCodes = $attribute->getLocaleSpecificCodes();
+                if (in_array($this->currentLocale, $availableCodes)) {
+                    if (isset($attributesToFilter[$attributeCode])) {
+                        $valuesToExport[$attributeCode] = $value;
+                    }
+                }
```

Comment:
Maybe you can use something like this ?

```suggestion
            if ($attribute->isLocaleSpecific()
                && in_array($this->currentLocale, $attribute->getLocaleSpecificCodes())
                && isset($attributesToFilter[$attributeCode])
            ) {
                $valuesToExport[$attributeCode] = $value;
            }
```